libcoap/src/protocol.rs

```diff
@@ -107,6 +93,8 @@ pub enum CoapOptionType {
     ProxyUri = COAP_OPTION_PROXY_URI as u16,
     /// Proxy-Scheme option ([RFC 7252, Section 5.10.2](https://datatracker.ietf.org/doc/html/rfc7252#section-5.10.2)).
     ProxyScheme = COAP_OPTION_PROXY_SCHEME as u16,
+    /// Observe option ([RFC 7641, Section 2](https://datatracker.ietf.org/doc/html/rfc7641#section-2)).
+    Observe = COAP_OPTION_OBSERVE as u16,
```

Comment:
I suppose the enum values are sorted by the number of the RFC they are defined in at the moment, right? I'd personally prefer a sorting by option number to have it aligned with the IANA registry, but that is just another matter of taste.

Reply:
Yes, currently the sorting is based on the RFC number.

I actually thought about using the IANA registry numbering while writing the code, but was a bit bothered by closely related options (such as Q-Block1 and Q-Block2) being separated.

However, as this would also result in a numeric ordering of the enum values, I'm going to switch over to the IANA numbering instead.

Reply:
Done. Options are now sorted by IANA numbering.